tests/lib/SimpleSAML/Utils/SystemTest.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+namespace SimpleSAML\Test\Utils;
+
+use \SimpleSAML_Configuration as Configuration;
+use \SimpleSAML\Utils\System;
+
+use \org\bovigo\vfs\vfsStream;
+
+/**
+ * Tests for SimpleSAML\Utils\System.
+ */
+class SystemTest extends \PHPUnit_Framework_TestCase
+{
+    const ROOTDIRNAME = 'testdir';
+
+    public function setUp()
+    {
+        $this->root = vfsStream::setup(
+            self::ROOTDIRNAME,
+            null,
+            array(
+                'tempdir' => array(),
+            )
+        );
+        $this->root_directory = vfsStream::url(self::ROOTDIRNAME);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::getOS
+     * @test
+     */
+    public function testGetOSBasic()
+    {
+        $res = System::getOS();
+
+        $this->assertInternalType("int", $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathRemoveTrailingSlashes()
+    {
+        $base = "/base////";
+        $path = "test";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/base/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathPreferAbsolutePathToBase()
+    {
+        $base = "/base/";
+        $path = "/test";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathCurDirPath()
+    {
+        $base = "/base/";
+        $path = "/test/.";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::resolvePath
+     * @test
+     */
+    public function testResolvePathParentPath()
+    {
+        $base = "/base/";
+        $path = "/test/child/..";
+
+        $res = System::resolvePath($path, $base);
+        $expected = "/test";
+
+        $this->assertEquals($expected, $res);
+    }
+
+    /**
+     * @covers \SimpleSAML\Utils\System::writeFile
+     * @test
+     */
+    public function testWriteFileInvalidArguments()
+    {
+        $this->setExpectedException('\InvalidArgumentException');
+        System::writeFile(null, null, null);
+    }
+
+    /**
+     * @requires PHP 5.4.0
+     * @covers \SimpleSAML\Utils\System::writeFile
+     * @test
+     */
+    public function testWriteFileBasic()
+    {
+        $tempdir = $this->root_directory . DIRECTORY_SEPARATOR . 'tempdir';
+        $config = Configuration::loadFromArray(array(
+            'tempdir' => $tempdir,
+        ), '[ARRAY]', 'simplesaml');
```

Comment:
I see this code is repeated in several test cases, with the last part of the directory being the only thing that changes from method to method. Wouldn't it be best to have a private function that initializes the configuration receiving the directory name?